### PR TITLE
feat(modal): provides `content-top` and `content-bottom` slots

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #4800

## Summary
Provides `content-top` and `content-bottom` slots. 

For simplicity, we matched the `padding` on the `content-top` and `content-bottom` to the `content`, with one public prop for the latter. This can further be developed if requested.

We decided to move forward with `content-top` and `content-bottom` naming, as this should provide enough context and also aligns with existing `<x>-start/<x>-end` naming.

### CODE REVIEW

File: src/components/modal/modal.scss
Comment (macandcheese) [inline src/components/modal/modal.scss:8]: cc @ashetland / @SkyeSeitz - slot names could be `content-top` and `content-bottom` to avoid confusion with existing elements.

Comment (ashetland) [reply src/components/modal/modal.scss:8]: Would `content-start` and `content-end` be more inline with existing conventions or do we reserve start/end for slots on the x-axis?

Comment (Elijbet) [reply src/components/modal/modal.scss:8]: What do you mean by this?

Comment (ashetland) [reply src/components/modal/modal.scss:8]: We use start/end in multiple places, for example icon-start/end and actions-end (I think we have actions-start somewhere?). So instead of naming these slots `content-top` and `content-bottom` would `content-start` and `content-end` make sense? Or would top/bottom be preferred here?

Comment (SkyeSeitz) [reply src/components/modal/modal.scss]: I don't think it would hurt to reserve start & end with the x-axis in case we ever needed both top/bottom & start/end, so `content-top` & `content-bottom` would be my vote unless there were additional concerns.

File: src/components/modal/modal.scss
Comment (jcfranco) [inline src/components/modal/modal.scss:207]: Is `z-header` needed for the top and bottom areas?

File: src/components/modal/modal.scss
Comment (jcfranco) [inline src/components/modal/modal.scss:203]: Is rounding needed for when there's no header nor footer?

File: src/components/modal/modal.scss
Comment (jcfranco) [inline src/components/modal/modal.scss:212]: Can you DRY up the common utils from both `.content-top` and `.content-bottom`?

File: src/components/modal/modal.scss
Comment (jcfranco) [inline src/components/modal/modal.scss:229]: Since the slots were renamed, can this go back to the previous name?

This applies to other modal-footer renames in this PR.

File: src/components/modal/modal.tsx
Comment (jcfranco) [inline src/components/modal/modal.tsx:269]: It looks like the content top and bottom nodes are always rendered, so these don't need `key` (see [Stencil doc on conditional rendering](https://stenciljs.com/docs/templating-jsx#:~:text=If%20this%20array%20of%20children%20is%20dynamic%2C%20i.e.%2C%20if%20any%20nodes%20may%20be%20added%2C%20removed%2C%20or%20reordered%2C%20then%20it%27s%20a%20good%20idea%20to%20set%20a%20unique%20key%20attribute%20on%20each%20element)).

**Review by jcfranco — APPROVED** [review]
Aside from a few comments, this LGTM!

**Comment by jcfranco** [comment]
@Elijbet Let's wait until the patch release is done before merging this.

File: src/components/modal/modal.scss
Comment (Elijbet) [reply src/components/modal/modal.scss:207]: It's on the same level as the header, so I kept it.

File: src/components/modal/modal.scss
Comment (Elijbet) [reply src/components/modal/modal.scss:203]: I made a correction here to have the top rounded on the `content-top` if there is no `header`, and the rounded bottom on the `content-bottom` if there is no `footer`.

**Pull request merged**
